src/txn/txn_rollback_to_stable.c

```diff
@@ -277,23 +319,34 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
          * update chain. Also it confirms that history store doesn't contains any newer version than
          * the current version for the key.
          */
-        if (!replace && hs_stop_durable_ts <= rollback_timestamp) {
+        if (!replace &&
+          (!__rollback_check_if_txnid_non_committed(session, cbt->upd_value->tw.stop_txn)) &&
+          (hs_stop_durable_ts <= rollback_timestamp)) {
             __wt_verbose(session, WT_VERB_RTS,
-              "history store update valid with stop timestamp: %s and stable timestamp: %s",
+              "history store update valid with stop timestamp: %s, stable timestamp: %s and txnid "
+              ": %" PRIu64,
               __wt_timestamp_to_string(hs_stop_durable_ts, ts_string[0]),
-              __wt_timestamp_to_string(rollback_timestamp, ts_string[1]));
+              __wt_timestamp_to_string(rollback_timestamp, ts_string[1]),
+              cbt->upd_value->tw.stop_txn);
             break;
         }
 
-        /* Stop processing when we find a stable update according to the given timestamp. */
-        if (hs_durable_ts <= rollback_timestamp) {
+        txn_non_committed =
```

Comment:
The `txn_non_committed` variable is used only here and in the rest of the places the macro is used directly. Use the macro itself here also instead of the variable.

Reply:
Fixed the unused variable.